fetch/range/blob.any.js

```diff
@@ -0,0 +1,100 @@
+// META: script=/common/utils.js
+
+promise_test(async () => {
+  var blob = new Blob(["A simple Hello, World! example"], { "type" : "text/plain" });
+  return fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=9-21"
+    }
+  }).then(function(resp) {
+    assert_equals(resp.status, 206, "HTTP status is 206");
+    assert_equals(resp.type, "basic", "response type is basic");
+    assert_equals(resp.headers.get("Content-Type"), "text/plain", "Content-Type is " + resp.headers.get("Content-Type"));
+    assert_equals(resp.headers.get("Content-Length"), "13", "Content-Length is " + resp.headers.get("Content-Length"));
+    assert_equals(resp.headers.get("Content-Range"), "bytes 9-21/30", "Content-Range is " + resp.headers.get("Content-Range"));
+    return resp.text();
+  }).then(function(text) {
+    assert_equals(text, 'Hello, World!', "Response's body is 'Hello, World!'");
+  });
+}, "Blob range request slices the blob");
+
+promise_test(function(test) {
+  var blob = new Blob(["Multiple ranges are not currently supported"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=0-5,15-"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with multiple range values");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range end should be greater than range start"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=10-5"
+    }
+  });
+  return promise_rejects_js(test, TypeError, promise);
+}, "Blob range request with short range end");
+
+promise_test(function(test) {
+  var blob = new Blob(["Range with no start or end"], { "type" : "text/plain" });
+  var promise = fetch(URL.createObjectURL(blob), {
+    "method": "GET",
+    "headers": {
+      "Range": "bytes=-"
```

Comment:
Not sure what to do here. I don't think this is valid according to [RFC 7233 Section 2.1](https://httpwg.org/specs/rfc7233.html#rfc.section.2.1)

Reply:
I think it ends up matching `other-range` (`-` being 0x2D), but that's indeed something we should treat as an error.